SDWebImage/SDImageCacheConfig.m

```diff
@@ -7,21 +7,59 @@
  */
 
 #import "SDImageCacheConfig.h"
+#import "SDMemoryCache.h"
+#import "SDDiskCache.h"
 
+static SDImageCacheConfig *_defaultCacheConfig;
 static const NSInteger kDefaultCacheMaxCacheAge = 60 * 60 * 24 * 7; // 1 week
 
 @implementation SDImageCacheConfig
 
++ (SDImageCacheConfig *)defaultCacheConfig {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        _defaultCacheConfig = [SDImageCacheConfig new];
+    });
+    return _defaultCacheConfig;
+}
+
++ (void)setDefaultCacheConfig:(SDImageCacheConfig *)defaultCacheConfig {
```

Comment:
I can't think of a case where you would want to set a default cache config. You can just pass one to all the initializers if you want to override. 
My original idea with this config object was that you can never change it, just edit the properties, to keep it simple.

I think this setter is confusing, since it's not obvious what "setting a default cache config" does under the hood.

Reply:
@bpoplauschi Anyway, I can change back to this design. Wait for a commit. (Also fix merge conflict)